src/protocol/ipa_prf/boolean_ops/share_conversion_aby.rs

```diff
@@ -0,0 +1,361 @@
+/// Propagates Errors from Interger Subtraction and Partial Reveal
+#[cfg(all(test, unit_test))]
+async fn convert_to_fp25519<C, B>(
+    ctx: C,
+    record_id: RecordId,
+    x: &AdditiveShare<B>,
+) -> Result<AdditiveShare<Fp25519>, Error>
+where
+    C: Context,
+    for<'a> &'a AdditiveShare<B>: IntoIterator<Item = AdditiveShare<B::Element>>,
+    B: WeakSharedValue + CustomArray<Element = Boolean> + Field,
+{
+    // we generate r <- (r1,r2,r3) and redefine r = (0,0,r3), and s = (r1,0,0)
+    let mut sh_r: AdditiveShare<BA256> = ctx
+        .narrow(&Step::GenerateSecretSharing)
+        .prss()
+        .generate_replicated(record_id);
+
+    // set 2 highest order bits to 0 to allow carries for two additions
+    sh_r.set(255, AdditiveShare::<Boolean>::ZERO);
+    sh_r.set(254, AdditiveShare::<Boolean>::ZERO);
+
+    // set r2=0
+    match ctx.role() {
+        Role::H1 => sh_r.1 = <BA256 as WeakSharedValue>::ZERO,
+        Role::H2 => sh_r.0 = <BA256 as WeakSharedValue>::ZERO,
+        Role::H3 => (),
+    }
+
+    //generate s from r
+    let mut sh_s = AdditiveShare::<BA256>::ZERO;
+
+    //redefine (r, s): H1: ((0,0),(r1,0)), H2: ((0,r3),(0,0)), H3: ((r3,0),(0,r1))
+    match ctx.role() {
+        Role::H1 => sh_r.0 = <BA256 as WeakSharedValue>::ZERO,
+        Role::H2 => sh_s.1 = <BA256 as WeakSharedValue>::ZERO,
+        Role::H3 => {
+            sh_r.1 = <BA256 as WeakSharedValue>::ZERO;
+            sh_s.0 = <BA256 as WeakSharedValue>::ZERO;
+        }
+    }
+
+    // addition r+s might cause carry, so we need 253 bits
+    let mut sh_rs = integer_add::<_, BA256, BA256>(
+        ctx.narrow(&Step::IntegerAddBetweenMasks),
+        record_id,
+        &sh_r,
+        &sh_s,
+    )
+    .await?;
+
+    //PRSS/Multiply masks added random highest order bit,
+    // remove them to not cause overflow in second addition (which is mod 256):
+    sh_rs.set(255, AdditiveShare::<Boolean>::ZERO);
+
+    //addition x+rs, where rs=r+s might cause carry
+    let sh_y =
+        integer_add::<_, BA256, B>(ctx.narrow(&Step::IntegerAddMaskToX), record_id, &sh_rs, x)
+            .await?;
+
+    // this leaks information, but with negligible probability
+    let y = AdditiveShare::<BA256>(sh_y.left(), sh_y.right())
+        .partial_reveal(ctx.narrow(&Step::RevealY), record_id, Role::H3)
+        .await?;
+
+    match ctx.role() {
+        Role::H1 => Ok(AdditiveShare::<Fp25519>(
+            Fp25519::from(sh_s.0).neg(),
+            Fp25519::from(y.unwrap()),
+        )),
+        Role::H2 => Ok(AdditiveShare::<Fp25519>(
+            Fp25519::from(y.unwrap()),
+            Fp25519::from(sh_r.1).neg(),
+        )),
+        Role::H3 => Ok(AdditiveShare::<Fp25519>(
+            Fp25519::from(sh_r.0).neg(),
+            Fp25519::from(sh_s.1).neg(),
+        )),
+    }
+}
+
+/// inserts a smaller array into a larger
+/// allows conversion between Boolean Array types like 'BA64' and 'BA256'
+/// we don't use it right except for testing purposes
```

Comment:
We don't use it _right_? What do you mean? Do we use it wrong?

Reply:
I don't like this comment. Do you mean:

/// we don't use it except for testing purposes

Because this sounds like you're saying:

/// we use this code incorrectly, except in our tests, where it is used correctly.

Reply:
right now, we only use it for testing purposes. I forgot to write "now". We will need this functionality later for a secure quicksort where we need to add a counter to make each element unique.